### ISSUE TITLE
Decompiler: Full output for AnalyzeStructuring emits goto to missing IL_010C label

### ISSUE THREAD

**Issue by richlander** [richlander/dotnet-inspect#1748]
## Defect

The decompiler reports `Full` fidelity for `ILInspector.DecompilerHarness.CorpusSensor::AnalyzeStructuring`, but the rendered method does not bind: a `catch` block emits `goto IL_010C; // leave` and the final C# body contains no `IL_010C:` label in scope.

This is a claimed-`Full` validity defect (`CS0159`), found by an adversarial pass and verified with the harness.

## Repro

From a clean checkout:

```bash
dotnet build src/dotnet-inspect -c Release -p:PublishAot=false
dotnet build tools/DecompilerHarness -c Release

dotnet run --project tools/DecompilerHarness -c Release -- \
  tools/DecompilerHarness/bin/Release/net11.0/decompiler-harness.dll \
  --validity-check --compile-cap 400 --max-examples 20
```

Observed validity output:

```text
COMPILE-CHECK over 1263 rendered methods (1171 Full, 92 Partial)

Syntactic validity (parse + statement legality — false-positive-free):
  Full malformed   : 0 (0.00% of Full) — the "claimed good but won't parse" set
  Partial malformed: 83 (90.22% of Partial) — expected: the diagnosed unsupported bits

Semantic binding (Full + syntactically-valid, capped at 400 bound):
  with a non-binding-noise error: 2 (0.50%)
  by code (binding-visibility codes already filtered out):
    CS0023: 1
    CS0159: 1

Semantic-defect examples (Full):
  ILInspector.DecompilerHarness.CorpusSensor::AnalyzeStructuring
    CS0159: No such label 'IL_010C' within the scope of the goto statement
```

The decompiled body includes:

```csharp
try
{
    IrPasses.Run(function, IrPasses.Default, context);
}
catch
{
    crashes += ((long)1);
    goto IL_010C; // leave
}
structured += ((long)diagnostics.Structured);
```

There is no matching `IL_010C:` label in the emitted C# method.

Source shape in `tools/DecompilerHarness/CorpusSensor.cs`:

```csharp
try
{
    IrPasses.Run(function, IrPasses.Default, context);
}
catch
{
    crashes++;
    continue;
}
```

IL evidence around the `catch`:

```text
IL_008A: ldloc.s      11
IL_008C: call         ILInspector.Decompiler.Pipeline.IrPasses::get_Default()
IL_0091: ldloc.s      13
IL_0093: call         ILInspector.Decompiler.Pipeline.IrPasses::Run(...)
IL_0098: leave.s      IL_00A2
IL_009A: pop
IL_009B: ldloc.1
IL_009C: ldc.i4.1
IL_009D: conv.i8
IL_009E: add
IL_009F: stloc.1
IL_00A0: leave.s      IL_010C
...
IL_010C: ldloc.s      10
IL_010E: callvirt     System.Collections.IEnumerator::MoveNext()
IL_0113: brtrue       IL_005A
```

## Expected

The output should either structure this as `continue;` to the enumerator loop, retain a valid label target, or degrade honestly below `Full` instead of emitting an unbound `goto`.

Duplicate search found no direct open issue for `AnalyzeStructuring IL_010C CS0159`.

**Comment by richlander** [comment]
Follow-up broader repro: raising the compile cap to 800 shows this is a repeated missing-label shape, not isolated to `AnalyzeStructuring`:

```text
Semantic binding (Full + syntactically-valid, capped at 800 bound):
  with a non-binding-noise error: 9 (1.12%)
  by code (binding-visibility codes already filtered out):
    CS0159: 15
    CS0023: 2

Semantic-defect examples (Full):
  ILInspector.DecompilerHarness.CorpusSensor::AnalyzeStructuring
    CS0159: No such label 'IL_010C' within the scope of the goto statement\n  ILInspector.DecompilerHarness.FidelityCheck::Run\n    CS0159: No such label 'IL_0163' within the scope of the goto statement\n  ILInspector.DecompilerHarness.FidelityCheck::Evaluate\n    CS0159: No such label 'IL_01C7' within the scope of the goto statement\n  ILInspector.DecompilerHarness.FidelityCheck::EmitInterface\n    CS0159: No such label 'IL_023F' within the scope of the goto statement\n  ILInspector.DecompilerHarness.Program::DisambiguateDump\n    CS0159: No such label 'IL_009A' within the scope of the goto statement\n  ILInspector.DecompilerHarness.TypeBindCheck::Run\n    CS0159: No such label 'IL_009C' within the scope of the goto statement\n```\n\nCommand used:\n\n```bash\ndotnet run --project tools/DecompilerHarness -c Release -- \\\n  tools/DecompilerHarness/bin/Release/net11.0/decompiler-harness.dll \\\n  --validity-check --compile-cap 800 --max-examples 50\n```

**Comment by richlander** [comment]
Claiming this #1687 burndown row as runner. Scope: fix or honestly degrade the Full output that emits gotos to missing retained labels (starting with AnalyzeStructuring), add focused coverage/evidence, and drive to PR/blocker/pivot.

**Comment by richlander** [comment]
Runner update: #1748 is now in review via #1793. Post-rebase evidence: focused InfiniteLoopStructuringTests passed, fast ILInspector.Decompiler.Tests passed (1613/1613), row validity boss dropped the CS0159 bucket to zero with only unrelated CS0023 remaining, and PR quick corpus matched pinned-gate tolerances. Cross-model adversarial review is running and will be summarized on the PR before readiness.

**Comment by richlander** [comment]
Runner update: #1748 / #1793 is Ready to merge. Branch is clean, CI is green, row validity boss has no CS0159 bucket, and adversarial findings were reconciled on the PR with the over-broad region-exit leave conversion fixed in 85d623f1. A redundant final Opus re-review is running as non-blocking follow-up.

**Comment by richlander** [comment]
Runner update: #1748 / #1793 is Ready to merge again after the final Opus blocking finding was fixed in 64c00698. Branch is clean, CI is green, focused/fast decompiler tests passed, row validity boss has no CS0159 bucket, and review reconciliation is posted on the PR.
